src/main/java/duke/Task.java

```diff
@@ -0,0 +1,57 @@
+package duke;
+
+import java.io.Serializable;
+
+/**
+ * The parent class of the specific tasks.
+ */
+public class Task implements Serializable {
+    private String name;
+    private boolean status;
+
+    /**
+     * The constructor of this class.
+     *
+     * @param name
+     */
+    public Task(String name) {
+        this.status = false;
+        this.name = name;
+    }
+
+    /**
+     * The method that modifies the status of task.
+     *
+     * @param status
+     */
+    public void setStatus(boolean status) {
```

Comment:
This method can be separated into two different methods. One only for Mark and the other only for Unmark. This is to be easy to read for the code readers (also a coding habit from CS2030S).